src/devices/cpu/i386/i386.cpp

```diff
@@ -285,15 +381,30 @@ bool i386_device::translate_address(int pl, int type, uint32_t *address, uint32_
 
 	if (!(entry & VTLB_FLAG_VALID) || ((type & TRANSLATE_WRITE) && !(entry & VTLB_FLAG_DIRTY)))
 	{
-		if (!i386_translate_address(type, address, &entry))
+		if (!(m_cr[4] & 0x00000020))
 		{
-			*error = ((type & TRANSLATE_WRITE) ? 2 : 0) | ((m_CPL == 3) ? 4 : 0);
-			if (entry)
-				*error |= 1;
-			return false;
+			if (!i386_translate_address_normal(type, address, &entry))
+			{
+				*error = ((type & TRANSLATE_WRITE) ? 2 : 0) | ((m_CPL == 3) ? 4 : 0);
+				if (entry)
+					*error |= 1;
+				return false;
+			}
+			vtlb_dynload(index, *address, entry);
+			return true;
+		}
+		else
+		{
+			if (!i386_translate_address_pae(type, address, &entry))
+			{
+				*error = ((type & TRANSLATE_WRITE) ? 2 : 0) | ((m_CPL == 3) ? 4 : 0);
+				if (entry)
+					*error |= 1;
+				return false;
+			}
+			vtlb_dynload(index, *address, entry);
+			return true;
 		}
-		vtlb_dynload(index, *address, entry);
-		return true;
```

Comment:
Why did you duplicate the `vtlb_dynload(index, *address, entry); return true;` and move it inside both branches of the `if`/`else`?  In fact, most of the code inside the `if`/`else` is duplicated.  One of the foundational principles of good programming is DRY.